pkg/eks/create.go

```diff
@@ -463,3 +474,158 @@ func getParameterValueFromOutput(key string, outputs []*cloudformation.Output) s
+	if err != nil {
+		return fmt.Errorf("could not create ebs csi driver role: %w", err)
+	}
+	if _, err := installEBSAddon(opts.EKSService, opts.Config, roleArn, opts.AddonVersion); err != nil {
+		return fmt.Errorf("failed to install ebs csi driver addon: %w", err)
+	}
+
+	return nil
+}
+
+func configureOIDCProvider(iamService services.IAMServiceInterface, eksService services.EKSServiceInterface, config *eksv1.EKSClusterConfig) (string, error) {
+	output, err := iamService.ListOIDCProviders(&iam.ListOpenIDConnectProvidersInput{})
+	if err != nil {
+		return "", err
+	}
+	clusterOutput, err := eksService.DescribeCluster(&eks.DescribeClusterInput{
+		Name: aws.String(config.Spec.DisplayName),
+	})
+	if err != nil {
+		return "", err
+	}
+	if clusterOutput == nil {
+		return "", fmt.Errorf("could not find cluster [%s]", config.Spec.DisplayName)
+	}
+	id := path.Base(*clusterOutput.Cluster.Identity.Oidc.Issuer)
+
+	for _, prov := range output.OpenIDConnectProviderList {
+		if strings.Contains(*prov.Arn, id) {
+			return "", nil
+		}
+	}
+
+	thumbprint, err := getIssuerThumbprint(*clusterOutput.Cluster.Identity.Oidc.Issuer)
+	if err != nil {
+		return "", err
+	}
+	input := &iam.CreateOpenIDConnectProviderInput{
+		ClientIDList:   []*string{aws.String(defaultAudienceOpenIDConnect)},
+		ThumbprintList: []*string{&thumbprint},
+		Url:            clusterOutput.Cluster.Identity.Oidc.Issuer,
+		Tags:           []*iam.Tag{},
+	}
+	newOIDC, err := iamService.CreateOIDCProvider(input)
+	if err != nil {
+		return "", err
+	}
+
+	return path.Base(*newOIDC.OpenIDConnectProviderArn), nil
+}
+
+func getIssuerThumbprint(issuer string) (string, error) {
+	issuerURL, err := url.Parse(issuer)
+	if err != nil {
+		return "", err
+	}
+	if issuerURL.Port() == "" {
+		issuerURL.Host += ":443"
+	}
+
+	client := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				InsecureSkipVerify: true,
+				MinVersion:         tls.VersionTLS12,
+			},
+			Proxy: http.ProxyFromEnvironment,
+		},
+	}
+	resp, err := client.Get(issuerURL.String())
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+
+	if resp.TLS == nil || len(resp.TLS.PeerCertificates) == 0 {
+		return "", err
+	}
+
+	root := resp.TLS.PeerCertificates[len(resp.TLS.PeerCertificates)-1]
+
+	return fmt.Sprintf("%x", sha1.Sum(root.Raw)), nil
+}
+
+func createEBSCSIDriverRole(cfService services.CloudFormationServiceInterface, config *eksv1.EKSClusterConfig, oidcID string) (string, error) {
+	templateData := struct {
+		Region     string
+		ProviderID string
+	}{
+		Region:     config.Spec.Region,
+		ProviderID: oidcID,
+	}
+	tmpl, err := template.New("ebsrole").Parse(templates.EBSCSIDriverTemplate)
+	if err != nil {
+		return "", err
+	}
+	buf := &bytes.Buffer{}
+	if execErr := tmpl.Execute(buf, templateData); execErr != nil {
+		return "", err
+	}
+	finalTemplate := buf.String()
+
+	output, err := CreateStack(CreateStackOptions{
```

Comment:
Would be great to not introduce any more CloudFormation....but understand why as all the other roles are created using CF

Reply:
I'm happy to move this to the SDK if that's better. I just used CloudFormation to align with other AWS resources created in the repo and so I could use the same removal process as well.